sqlmesh/dbt/adapter.py

```diff
@@ -209,6 +216,14 @@ def get_missing_columns(
             if col.name not in target_columns
         ]
 
+    def rename_relation(self, from_relation: BaseRelation, to_relation: BaseRelation) -> None:
+        if from_relation.identifier is not None and to_relation.identifier is not None:
+            new_name = to_relation.identifier
+            old_table = exp.table_(
+                from_relation.identifier, db=from_relation.schema, catalog=from_relation.database
+            )
+            self.engine_adapter.rename_table(old_table, new_name)
```

Comment:
Does this actually work? Will sqlmesh detect that it needs to recreate old_table?